types/ui/draggable.d.ts

```diff
@@ -0,0 +1,98 @@
+/**
+ * A UI utility to make an element draggable.
+ */
+declare class Draggable<R extends boolean | undefined = undefined> {
```

Comment:
I don't think the behavior is correct if a boolean that is not known at compile time is passed as `resizable`. If I understand correctly, currently the type of `resizeDown` would be inferred as `['mousedown', (e: Event) => void, false]` (etc.) because `boolean` does not extend `false | undefined`. But actually, the type should probably be `undefined | ['mouseup', (e: Event) => void, false]` (etc.) because we do not know which one it is at compile time.

There should also be a test case for this. I.e. something like
```ts
const someBoolean: boolean = false;
const maybeResizable = new Draggable(new Application(), $(), new HTMLElement(), someBoolean);
expectType<undefined | ['mousedown', (e: Event) => void, false]>(resizable.handlers.resizeDown);
expectType<undefined | ['mousemove', (e: Event) => void, false]>(resizable.handlers.resizeMove);
expectType<undefined | ['mouseup', (e: Event) => void, false]>(resizable.handlers.resizeUp);
```

Reply:
Alright, I improved the tests and change the type inference. This should be a lot better now.